connectors/sources/jira.py

```diff
@@ -216,65 +220,92 @@ def get_default_configuration(cls):
             dictionary: Default configuration.
         """
         return {
-            "is_cloud": {
-                "value": True,
-                "label": "True if Jira Cloud, False if Jira Server",
-                "type": "bool",
+            "data_source": {
```

Comment:
I remember @danajuratoni had a nice suggestion regarding this in design document: is it possible to control which fields to show based on the value of `data_source` ?
Since Server uses `username` and `password`, when `data_source` is Server, then `api_token` and `service_account_id` are not visible or disabled and vice-versa for Cloud.

Reply:
Yes that's what the `depends_on` field is for. If the `data_source` value is `jira_online` then only Jira Cloud inputs will appear below it. Vice versa exists for `jira_server`.